src/axom/primal/operators/clip.hpp

```diff
@@ -304,6 +304,82 @@ AXOM_HOST_DEVICE Polyhedron<T, 3> clip(const Tetrahedron<T, 3>& tet1,
   return detail::clipTetrahedron(tet1, tet2, eps, checkSign);
 }
 
+/*!
+ * \brief Clips a 3D tetrahedron against a plane in 3D, returning
+ *        the geometric intersection as a polyhedron
+ *
+ *  This function clips the tetrahedron against the given plane. This
+ *  involves finding new vertices at the intersection of the polyhedron
+ *  edges and the plane, removing vertices from the polyhedron that are
+ *  below the plane, and redefining the neighbors for each vertex (a
+ *  vertex is a neighbor of another vertex if there is an edge between
+ *  them).
```

Comment:
@adayton1 -- it might be clearer to say that this clips a 3D tetrahedron against the half-space defined by a plane

Reply:
Done.